api/src/services/users/create_api_key.py

```diff
@@ -59,6 +81,50 @@ def create_api_key(db_session: db.Session, user_id: UUID, json_data: dict) -> Us
     return api_key
 
 
+def _import_api_key_to_aws_gateway(api_key: UserApiKey) -> None:
+    """Import an API key to AWS API Gateway and associate it with a usage plan"""
+    try:
+        config = ApiGatewayConfig()
+
+        gateway_response = import_api_key(
+            api_key=api_key.key_id,
+            name=api_key.key_name,
+            description=f"API key for user {api_key.user_id}",
+            enabled=api_key.is_active,
+            usage_plan_id=config.default_usage_plan_id,
+        )
+
+        if config.default_usage_plan_id:
+            logger.info(
+                "Successfully imported API key to AWS API Gateway and associated with usage plan",
+                extra={
+                    "api_key_id": api_key.api_key_id,
+                    "gateway_key_id": gateway_response.id,
+                    "usage_plan_id": config.default_usage_plan_id,
+                },
+            )
+        else:
+            logger.info(
+                "Successfully imported API key to AWS API Gateway (no usage plan configured)",
+                extra={
+                    "api_key_id": api_key.api_key_id,
+                    "gateway_key_id": gateway_response.id,
+                },
+            )
+
+    except Exception as e:
+        logger.exception(
+            "Error importing API key to AWS API Gateway",
+            extra={
+                "api_key_id": api_key.api_key_id,
+                "key_name": api_key.key_name,
+            },
+        )
+        raise ApiGatewayIntegrationError(
+            f"Failed to import API key to AWS API Gateway: {str(e)}"
+        ) from e
```

Comment:
If we have an except that logs this in the library, and above this, do we need this try-except?